compiler/src/iree/compiler/Codegen/Transforms/Transforms.h

```diff
@@ -23,16 +24,33 @@
 namespace mlir {
 namespace iree_compiler {
 
-/// Insert patterns to perform folding of AffineMinOp by matching the pattern
-/// generated by tile and distribute. Try to fold a affine.min op by matching
-/// the following form:
+/// Get the `offsets`, `sizes` and `strides` for a `storeOp` (or `loadOp`). This
+/// method clones the operations that generate the `Value`s used for
+/// specifying the offsets, sizesm strides and dynamic dims of the
+/// `storeOp/loadOp` at the insertion point to avoid use-def violations.
+void cloneOffsetsSizesAndStrides(OpBuilder &builder,
+                                 IREE::Flow::DispatchTensorStoreOp storeOp,
+                                 SmallVector<OpFoldResult> &offsets,
+                                 SmallVector<OpFoldResult> &sizes,
+                                 SmallVector<OpFoldResult> &strides,
+                                 SmallVector<Value> &dynamicDims);
+void cloneOffsetsSizesAndStrides(OpBuilder &builder,
+                                 IREE::Flow::DispatchTensorLoadOp loadOp,
+                                 SmallVector<OpFoldResult> &offsets,
+                                 SmallVector<OpFoldResult> &sizes,
+                                 SmallVector<OpFoldResult> &strides,
+                                 SmallVector<Value> &dynamicDims);
+
```

Comment:
should we move these to https://github.com/iree-org/iree/blob/main/compiler/src/iree/compiler/Codegen/Utils/Utils.h ?

Reply:
I'd rather keep these in `Transforms` cause it is actually changing code... I think `Utils` is just for analysis/querrying methods.

Reply:
I was confused about what to put into `Utils`. Maybe we can rename it to `AnalysisUtils.h`, I can do the cleanup.